app/models/chapter.rb

```diff
@@ -45,4 +45,36 @@ def display_with_mlid
   def full_mlid
     "#{organization.mlid}-#{mlid}"
   end
+
+  def delete_chapter_and_dependents
```

Comment:
Since this method is called `delete...` it should actually perform deletion rather than just setting the `deleted_at` field and not saving itself to the db. Let's call `save` at the end of the transaction but still inside transaction.